packages/near-api-js/src/account.ts

```diff
@@ -710,4 +711,100 @@ export class Account {
             total: summary.total.toString(),
         };
     }
+
+    createTransaction(recevier: Account | string): Transaction {
+        return new Transaction(this, recevier);
+    }
+  
+    async hasDeployedContract(): Promise<boolean> {
+        return (await this.state()).code_hash !== EMPTY_CONTRACT_HASH;
+    }
 }
+
+
+/**
+ * Transaction Builder class. Initialized to an account that will sign the final transaction
+ */
+export class Transaction {
```

Comment:
Is there a compelling reason to not break this out into its own module instead of exporting it from `account.ts`? This file is already absurdly long and if other code is going to import this then it makes more sense for it to have its own module.

Reply:
My original intent was that there was a interdependency between the two classes. An Account can create a txn builder, which in turn requires a signer (an Account).  But perhaps it really just requires a signer and the account can pass the signer.

```ts
this.signer.signAndSendTransaction({ receiverId: this.receiverId, actions: this.actions })
this.signer.signTransaction(this.receiverId, this.actions);
```

Reply:
Yeah I think constructing these with signer makes a lot of sense and breaks the require cycle. Moving this class into it's own module 👍

Reply:
I've pulled the access key caching and transaction sending functionality into a new `TransactionSender` class that both `Account` and `TransactionBuilder` extend